build_process/mocks/compile_one_diagram/output.json

```diff
@@ -0,0 +1,61 @@
+{
+    "kindName": "default-component-name",
+    "parts": [
+        {
+            "inCount": 1,
+            "inMap": {
+                "in": 0
+            },
+            "inPins": [
+                [
+                    1
+                ]
+            ],
+            "kindName": "part",
+            "outCount": 1,
+            "outMap": {
+                "out": 0
+            },
+            "outPins": [
+                [
+                    0
+                ]
+            ],
+            "partName": "ID12"
+        },
+        {
+            "inCount": 1,
+            "inMap": {
+                "source": 0
+            },
+            "inPins": [
+                [
+                    0
+                ]
+            ],
+            "kindName": "self",
+            "outCount": 0,
+            "outMap": {},
+            "outPins": [],
+            "partName": "ID11"
+        },
+        {
+            "inCount": 0,
+            "inMap": {},
+            "inPins": [],
+            "kindName": "self",
+            "outCount": 1,
+            "outMap": {
+                "sink": 0
+            },
+            "outPins": [
+                [
+                    1
+                ]
+            ],
+            "partName": "ID8"
+        }
+    ],
+    "self": {},
```

Comment:
@guitarvydas For whatever reason, the compiler doesn't generate `self` but the wiring is correct in `parts`. The two parts with the `kindName` of `self` are supposed to be a source and a sink. Do you know what could be the cause?

This is what the schematic looks like. Ignore the "name" texts and the dashed rectangle. I'm using a working copy of the editor so that part doesn't work.

<img width="1208" alt="Arrowgram Editor 2019-12-07 23-29-38" src="https://user-images.githubusercontent.com/1072977/70384222-8f8e5c00-1949-11ea-89e6-82a3859c0dd3.png">